protoc_gen_d/protoc-gen-d.d

```diff
@@ -146,9 +150,18 @@ class CodeGenerator
         if (messageType.isMap)
             return "";
 
+        auto strIndent = "%*s".format(indent, "");
+
         auto result = appender!string;
-        result ~= "\n%*s%sclass %s\n".format(indent, "", indent > 0 ? "static " : "", messageType.name.escapeKeywords);
-        result ~= "%*s{\n".format(indent, "");
+        result ~= "\n";
+        result ~= strIndent;
+        result ~= indent > 0 ? "static " : "";
+        result ~= makeStructs ? "struct" : "class";
```

Comment:
Please check here if the protobuf message definition is recursive. `struct` cannot be used for recursive messages.
This should be check in the code generator, otherwise the user would need to deal with it at compile time, probably not knowing that the generated message definition is wrong.

Reply:
What you mean by recursive messages? `A` in `B` as field, and `B` in `A` as field, right?
Fields must be pointers in this case and there comes more difficult code for working with its (get address of stack variable is bad idea, need use GC for allocate memory, or any other variant). I think this case is example where structs are not suits.